tests/test_mdlp.py

```diff
@@ -32,165 +34,52 @@ def test_find_cut_no_cut():
     k = find_cut(y, 0, len(y))
     assert_equal(-1, k)
 
-def test_mdlp_iris():
-    from sklearn.datasets import load_iris
+def test_e2e():
+  X = np.array([
+    [0.1, 0.1],
+    [0.2, 1.1],
+    [0.3, 0.2],
+    [1.5, .9]
+  ])
+  y = np.array([0, 0, 0, 1])
+  transformed = MDLP(shuffle=False).fit_transform(X, y)
+  expected = np.array([
+    [0, 0],
```

Comment:
This is a bit of a precarious test case. Everything seems to discretize to zero. Can you come up with a test case with a more variety of outputs?

Reply:
The two columns display both successful and unsuccessful discretization of data. It seems valuable to have a failure case here that discretizes to all 0s. The first column is not all 0 in this test case already but I will modify to make it even more interesting.